wagtail/admin/tests/tests.py

```diff
@@ -333,15 +333,36 @@ def test_no_permissions(self):
 class Test404(TestCase, WagtailTestUtils):
     def test_admin_404_template_used(self):
         self.login()
-        response = self.client.get('/admin/sdfgdsfgdsfgsdf')
+        response = self.client.get('/admin/sdfgdsfgdsfgsdf', follow=True)
+
+        # Check 400 error after CommonMiddleware redirect
         self.assertEqual(response.status_code, 404)
         self.assertTemplateUsed(response, 'wagtailadmin/404.html')
 
     def test_not_logged_in_redirect(self):
-        response = self.client.get('/admin/sdfgdsfgdsfgsdf')
```

Comment:
Otherwise it redirects first to `/admin/sdfgdsfgdsfgsdf/` and then `reverse('wagtailadmin_login') + '?next=/admin/sdfgdsfgdsfgsdf/`